clang/lib/Parse/ParseExprCXX.cpp

```diff
@@ -3030,23 +3029,13 @@ bool Parser::ParseUnqualifiedId(CXXScopeSpec &SS, ParsedType ObjectType,
           SS, ObjectType, ObjectHadErrors,
           TemplateKWLoc ? *TemplateKWLoc : SourceLocation(), Id, IdLoc,
           EnteringContext, Result, TemplateSpecified);
+    else if (TemplateSpecified &&
```

Comment:
These revert code do not have `SYCLomatic` tag, and will conflict once llvm or sycl touch these code. Can we just modify the core logic that cause the problem?

Reply:
`SYCLomatic_CUSTOMIZATION` is needed here to mark the difference with clang main trunk.